src/Api/FingerprintApi.php

```diff
@@ -117,70 +119,43 @@ public function getEvent(string $request_id): array
-            } catch (SerializationException $e) {
-                $e->setResponse($response);
-
-                throw $e;
-            }
+            $serialized = ObjectSerializer::deserialize($response, $returnType);
 
             return [$serialized, $response];
         } catch (ApiException $e) {
-            try {
-                switch ($e->getCode()) {
-                    case 200:
-                        /** @var ResponseInterface $response */
-                        $response = $e->getResponseObject();
-                        $responseBody = $response->getBody()->getContents();
-                        $response->getBody()->rewind();
-                        $data = ObjectSerializer::deserialize(
-                            $responseBody,
-                            '\Fingerprint\ServerAPI\Model\EventResponse',
-                            $response->getHeaders()
-                        );
-                        $e->setResponseObject($data);
-
-                        break;
-
-                    case 403:
-                        /** @var ResponseInterface $response */
-                        $response = $e->getResponseObject();
-                        $responseBody = $response->getBody()->getContents();
-                        $response->getBody()->rewind();
-                        $data = ObjectSerializer::deserialize(
-                            $responseBody,
-                            '\Fingerprint\ServerAPI\Model\ErrorEvent403Response',
-                            $response->getHeaders()
-                        );
-                        $e->setResponseObject($data);
-
-                        break;
-
-                    case 404:
-                        /** @var ResponseInterface $response */
-                        $response = $e->getResponseObject();
-                        $responseBody = $response->getBody()->getContents();
-                        $response->getBody()->rewind();
-                        $data = ObjectSerializer::deserialize(
-                            $responseBody,
-                            '\Fingerprint\ServerAPI\Model\ErrorEvent404Response',
-                            $response->getHeaders()
-                        );
-                        $e->setResponseObject($data);
-
-                        break;
-                }
-
-                throw $e;
-            } catch (SerializationException $exception) {
-                $exception->setResponse($e->getResponseObject());
-
-                throw $exception;
+            /** @var ResponseInterface $response */
+            $response = $e->getResponseObject();
+
+            switch ($e->getCode()) {
+                case 200:
+                    $data = ObjectSerializer::deserialize(
+                        $response,
+                        '\Fingerprint\ServerAPI\Model\EventResponse'
+                    );
+                    $e->setResponseObject($data);
```

Comment:
It's not a correct behavior. `ApiException` expects `$responseObject` to be `ResponseInterface`, but here we set parsed response body. Let's introduce another property to the exception that will store the parsed body instead.

Reply:
Done and done 👍